### PR TITLE
Parse chunk shape to check for float values

### PR DESCRIPTION
**Pull request by faymanns**
This PR solves the fact that specifying a chunk shape containing one or several float values does not raise an error.
Before the PR the values after the decimal point are trimmed without a warning.
After the PR a `TypeError` is raised when the chunk shape contains float values.
It solves issue #2530.

TODO:
* [x] Add unit tests and/or doctests in docstrings
* [ ] Add docstrings and API docs for any new/modified user-facing classes and functions
* [ ] New/modified features documented in docs/tutorial.rst
* [ ] Changes documented in docs/release.rst
* [ ] GitHub Actions have all passed
* [ ] Test coverage is 100% (Codecov passes)

### CODE REVIEW

**Review by d-v-b — APPROVED** [review]
looks good, thank you

File: src/zarr/core/array.py [hunk 1]
```diff
@@ -461,8 +461,11 @@ async def create(
             raise ValueError("Only one of chunk_shape or chunks can be provided.")
 
         if chunks:
+            chunks = parse_shapelike(chunks)
             _chunks = normalize_chunks(chunks, shape, dtype_parsed.itemsize)
         else:
+            if chunk_shape:
+                chunk_shape = parse_shapelike(chunk_shape)
```
Comment (jhamman) [inline src/zarr/core/array.py:468]: I suggest we move this check into `normalize_chunks`. Here's where the behavior you're looking to change is:

https://github.com/zarr-developers/zarr-python/blob/2fe12a7f424ac78c18a688d0a75049a3df67c98f/src/zarr/core/chunk_grids.py#L116-L117

If you look at `normalize_chunks`, you'll see why Zarr is more permissive on chunk shape than array shape. I'm not opposed to us disallowing floats but we need to be careful keep some of the other behaviors in `normalize_chunks`.

**Review by jhamman — CHANGES_REQUESTED** [review]
Thanks @faymanns for working on this. I left one comment to direct your attention to `normalize_chunks`

File: src/zarr/core/array.py (hunk 1 above)
Comment (faymanns) [reply src/zarr/core/array.py]: That seems like a better place indeed! I've updated the PR.

**Review by jhamman — APPROVED** [review]

**Pull request merged**